wgpu/src/util/indirect.rs

```diff
@@ -0,0 +1,55 @@
+/// The structure expected in `indirect_buffer` for [`RenderEncoder::draw_indirect`](crate::util::RenderEncoder::draw_indirect).
+#[repr(C)]
+#[derive(Copy, Clone, Debug, Default)]
+pub struct DrawIndirect {
+    /// The number of vertices to draw.
+    pub vertex_count: u32,
+    /// The number of instances to draw.
+    pub instance_count: u32,
+    /// The Index of the first vertex to draw.
+    pub base_vertex: u32,
+    /// The instance ID of the first instance to draw.
+    /// Has to be 0, unless [`Features::INDIRECT_FIRST_INSTANCE`](crate::Features::INDIRECT_FIRST_INSTANCE) is enabled.
+    pub base_instance: u32,
+}
+
+impl DrawIndirect {
+    /// Returns the bytes representation of the struct, ready to be written in a [`Buffer`](crate::Buffer).
+    pub fn as_bytes(&self) -> &[u8] {
+        unsafe {
+            std::mem::transmute(std::slice::from_raw_parts(
+                self as *const _ as *const u8,
+                std::mem::size_of::<Self>(),
+            ))
+        }
+    }
+}
+
+/// The structure expected in `indirect_buffer` for [`RenderEncoder::draw_indexed_indirect`](crate::util::RenderEncoder::draw_indexed_indirect).
+#[repr(C)]
+#[derive(Copy, Clone, Debug, Default)]
+pub struct DrawIndexedIndirect {
+    /// The number of vertices to draw.
+    pub vertex_count: u32,
+    /// The number of instances to draw.
+    pub instance_count: u32,
+    /// The base index within the index buffer.
+    pub base_index: u32,
+    /// The value added to the vertex index before indexing into the vertex buffer.
+    pub vertex_offset: i32,
+    /// The instance ID of the first instance to draw.
+    /// Has to be 0, unless [`Features::INDIRECT_FIRST_INSTANCE`](crate::Features::INDIRECT_FIRST_INSTANCE) is enabled.
+    pub base_instance: u32,
+}
+
+impl DrawIndexedIndirect {
+    /// Returns the bytes representation of the struct, ready to be written in a [`Buffer`](crate::Buffer).
+    pub fn as_bytes(&self) -> &[u8] {
+        unsafe {
+            std::mem::transmute(std::slice::from_raw_parts(
+                self as *const _ as *const u8,
```

Comment:
may I suggest: `<*const u8>::cast(self)`

Reply:
I don't like it mush either, but `<*const u8>::cast(self)` gives me:
```
mismatched types

expected `u8`, found struct `util::indirect::DrawIndirect`

note: expected raw pointer `*const u8`
           found reference `&util::indirect::DrawIndirect`
```